cms/djangoapps/contentstore/management/commands/backfill_orgs_and_org_courses.py

```diff
@@ -0,0 +1,168 @@
+
+        # Start with a dry run.
+        # This will log to the user which orgs/org-courses will be created/reactivated.
+        organizations_api.bulk_add_organizations(
+            orgs,
+            dry_run=True,
+        )
+        organizations_api.bulk_add_organization_courses(
+            org_coursekey_pairs,
+            dry_run=True,
+        )
+        if not should_apply_changes(options):
+            print("No changes applied.")
+            return
+
+        # It's go time.
+        print("Applying changes...")
+        organizations_api.bulk_add_organizations(
+            orgs,
+            dry_run=False,
+        )
+        organizations_api.bulk_add_organization_courses(
+            org_coursekey_pairs,
+            dry_run=False,
+        )
+        print("Changes applied successfully.")
+
+
+def should_apply_changes(options):
+    """
+    Should we apply the changes to the db?
+
+    If already specified on the command line, use that value.
+
+    Otherwise, prompt.
+    """
+    if options.get('apply') and options.get('dry'):
+        raise CommandError("Only one of 'apply' and 'dry' may be specified")
+    if options.get('apply'):
+        return True
+    if options.get('dry'):
+        return False
+    answer = ""
+    while answer.lower() not in {'y', 'yes', 'n', 'no'}:
+        answer = input('Commit changes shown above to the database [y/n]? ')
+    return answer.lower().startswith('y')
+
+
+def find_orgslug_coursekey_pairs():
+    """
+    Returns the unique pairs of (organization short name, course run key)
+    from the CourseOverviews table, which should contain all course runs in the
+    system.
+
+    Returns: set[tuple[str, CourseKey]]
+    """
+    # Using a set comprehension removes any duplicate (org, course) pairs.
+    return {
+        (course_key.org, course_key)
+        for course_key
+        # Worth noting: This will load all CourseOverviews, no matter their VERSION.
+        # This is intentional: there may be course runs that haven't updated
+        # their CourseOverviews entry since the last schema change; we still want
+        # capture those course runs.
+        in CourseOverview.objects.all().values_list("id", flat=True)
+    }
+
+
+def find_orgslug_library_pairs():
+    """
+    Returns the unique pairs of (organization short name, content library key)
+    from the modulestore.
+
+    Note that this only considers "version 1" (aka "legacy" or "modulestore-based")
+    content libraries.
+    We do not consider "version 2" (aka "blockstore-based") content libraries,
+    because those require a database-level link to their authoring organization,
+    and thus would not need backfilling via this command.
+
+    Returns: set[tuple[str, LibraryLocator]]
+    """
+    # Using a set comprehension removes any duplicate (org, library) pairs.
+    return {
+        (library_key.org, library_key)
+        for library_key
+        in modulestore().get_library_keys()
```

Comment:
Same question here.

Reply:
@davidjoy  While it's a harder for me to make predictions about Modulestore queries than SQL queries, I think this is an approximation of the query that'll be run:
```
db.modulestore.active_versions.find({
    "versions.library": {"$exists": true}
}).map(
    function(index) { return "library-v1:" + index.org + "+" + index.course }
)
```
On the read replica, this runs almost instantaneously for me (it generates a list of 7620 library keys).

Reply:
👍 At that order of magnitude I'd imagine it's fine.  If we said we had millions of them or something, then I'd start to be concerned. :)